contrib/net/http/httptrace.go

```diff
@@ -0,0 +1,127 @@
+package http
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"net/http/httptrace"
+	"time"
+)
+
+type httpTraceResult struct {
+	DNSLookup        time.Duration
+	TCPConnection    time.Duration
+	TLSHandshake     time.Duration
+	ServerProcessing time.Duration
+	NameLookup       time.Duration
+	Connect          time.Duration
+	Pretransfer      time.Duration
+	StartTransfer    time.Duration
+
+	dnsStart      time.Time
+	dnsDone       time.Time
+	tcpStart      time.Time
+	tcpDone       time.Time
+	tlsStart      time.Time
+	tlsDone       time.Time
+	serverStart   time.Time
+	serverDone    time.Time
+	transferStart time.Time
+
+	isTLS    bool
+	isReused bool
+
+	remoteIP   string
+	remotePort string
+}
+
+func WithClientTrace(ctx context.Context, r *httpTraceResult) context.Context {
+	return httptrace.WithClientTrace(ctx, &httptrace.ClientTrace{
```

Comment:
Can we please remove all the empty lines in this function? It's uncommon in Go to write code like this and generally they rarely occur. You can observe this practice in the standard library as well and throughout the repo. There are of course exceptions.

Reply:
Also not fully resolved.